File: python3/majority_element.py
# leetcode

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        leng = len(nums)
        if leng % 2 == 1:
            mid = (leng // 2) + 1
        else:
            mid = leng // 2
            
        mem = {}
        
        for i in nums:
            if i in mem:
                mem[i] += 1
            else:
                mem[i] = 1
        
        ans = 0
        for key, val in mem.items():
            if val >= mid:
                ans = key

        return ans

File: golang/majority_element.go
package main

/*
Space	: O(n)
Time	: O(n)
*/

func majorityElement(nums []int) int {
	ans := 0
	hashmap := make(map[int]int)

	for _, x := range nums {
		hashmap[x]++
	}

	count := 0
	for k, v := range hashmap {
		if v > count {
			count = v
			ans = k
		}
	}

	return ans
}
